Backend/lib/location-updates/src/Lib/LocationUpdates/Internal.hs

```diff
@@ -180,17 +184,20 @@ interpolatePointsAndCalculateDistanceImplementation ::
   ( HasCallStack,
     EncFlow m r,
     Metrics.CoreMetrics m,
-    HasField "snapToRoadSnippetThreshold" r HighPrecMeters
+    HasField "snapToRoadSnippetThreshold" r HighPrecMeters,
+    HasFlowEnv m r '["kafkaProducerTools" ::: KafkaProducerTools],
+    HasFlowEnv m r '["appPrefix" ::: Text]
   ) =>
   Bool ->
   MapsServiceConfig ->
+  Id person ->
   [LatLong] ->
   m (HighPrecMeters, [LatLong])
-interpolatePointsAndCalculateDistanceImplementation isEndRide mapsCfg wps = do
+interpolatePointsAndCalculateDistanceImplementation isEndRide mapsCfg someId wps = do
   if isEndRide && isAllPointsEqual wps
     then pure (0, take 1 wps)
     else do
-      res <- Maps.snapToRoad mapsCfg $ Maps.SnapToRoadReq {points = wps}
+      res <- Maps.snapToRoad (getId someId) mapsCfg $ Maps.SnapToRoadReq {points = wps}
```

Comment:
`someId` must be part of req

Reply:
Why? This id is used to form the kafka key